pkg/controller/argocd/policyrule.go

```diff
@@ -8,6 +8,19 @@ import (
 func policyRuleForApplicationController() []v1.PolicyRule {
 
 	return []v1.PolicyRule{
+		{
```

Comment:
On openshift, controller sa already has cluster admin https://github.com/argoproj-labs/argocd-operator/blob/master/pkg/reconciler/openshift/openshift.go#L48.   We should clean this up a bit to remove the delicated rules.  Here is the cluster admin role rules.  https://gist.githubusercontent.com/wtam2018/41f82ccee071cc3dc0f78b2d49805344/raw/ea09491d662747e16c57d7d43cebb7284fd092bb/gistfile1.txt

Reply:
This hook (https://github.com/argoproj-labs/argocd-operator/blob/master/pkg/reconciler/openshift/openshift.go#L48) binds the Cluster Role `admin` to a role binding which provides the `application-controller` `admin` access to the namespace it is installed in but not cluster-wide. 
while the rule defined here is for any ArgoCD instance irrespective of the cluster i.e Openshift or Non-Openshift will provide the access to get/list all resources for all API groups. 
Although I do agree, these roles are all over the place and we do need to look into it.

Reply:
@shubhamagarwal19 My point is  that on openshift hook, we already create cluster admin role (yes namespace scoped RoleBinding) bound to controller sa, policyRuleForApplicationController() is adding rules that already exist in cluster admin roles and we should remove duplicates here in policyRuleForApplicationController().

Reply:
Gottcha, but what I mean is that the rules defined in `policyRuleForApplicationController()` are for any ArgoCD instance irrespective of the Cluster it is deployed on. 
The hooks are defined for OpenShift specific deployments and yes, it will have the admin privileges for the namespace, but if anyone is using the ArgoCD operator on a non Openshift Cluster, we would need these rules. 
Am i missing something here ?

Reply:
You are right.  I thought `policyRuleForApplicationController` was invoked by the reconcileHook.  Nevermind then.

Reply:
We should move the openshift hook to openshift operator as a future cleanup.